src/redux/reducer.ts

```diff
@@ -848,26 +839,20 @@ export const reducer = createReducer(initialState, (builder) => {
     });
 
     builder.addCase(UPDATE_TABLE_DEFINITION, (state, action: UpdateTableDefinitionAction) => {
-        const { key, value, customColumns } = action.payload;
-        const updatedDefinitions = {
-            ...state.tables.definitions,
-            [key]: value,
-        };
-        const updatedColumnsNames = Object.values(updatedDefinitions)
-            .map((table) => table.columns)
+        const { newTableDefinition, customColumns } = action.payload;
+        const updatedDefinitions = [...state.tables.definitions];
+        updatedDefinitions.push(newTableDefinition as Draft<SpreadsheetTabDefinition>);
+        const updatedColumnsNames = updatedDefinitions
+            .map((tabDef) => tabDef.columns)
             .map((cols) => new Set(cols.map((c) => c.id)));
         const updatedColumnsNamesJson = updatedColumnsNames.map((cols) => JSON.stringify([...cols]));
-        const updatedNames = Object.values(updatedDefinitions).map((table) => table.name);
-        const updatedNamesIndexes = new Map(
-            Object.values(updatedDefinitions).map((table) => [table.name, table.index])
-        );
-        const updatedDefinitionTypes = new Map(Object.values(updatedDefinitions).map((table) => [table.type, table]));
-        const updatedDefinitionIndexes = new Map(
-            Object.values(updatedDefinitions).map((table) => [table.index, table])
-        );
+        const updatedNames = updatedDefinitions.map((tabDef) => tabDef.name);
+        const updatedNamesIndexes = new Map(updatedDefinitions.map((tabDef) => [tabDef.name, tabDef.index]));
+        const updatedDefinitionTypes = new Map(updatedDefinitions.map((tabDef) => [tabDef.type, tabDef]));
+        const updatedDefinitionIndexes = new Map(updatedDefinitions.map((tabDef) => [tabDef.index, tabDef]));
```

Comment:
Maybe deduplicate these functions code with config-tables.
Keeping code only here for constants not reuse elsewhere also?  
Also maybe use said functions in the initial state instead of constants?

Reply:
ok